float64 omega_l
float64 omega_r
